framework/py/flwr/common/object_store/in_memory_object_store.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower in-memory ObjectStore implementation."""
+
+from typing import Optional
+
+from ..inflatable import get_object_id, is_valid_sha256_hash
+from .object_store import ObjectStore
+
+
+class InMemoryObjectStore(ObjectStore):
+    """In-memory implementation of the ObjectStore interface."""
+
+    def __init__(self, verify: bool = True) -> None:
+        self.verify = verify
+        self.store: dict[str, bytes] = {}
+
+    def put(self, object_id: str, object_content: bytes) -> None:
+        """Put an object into the store."""
+        # Verify object ID format (must be a valid sha256 hash)
+        if not is_valid_sha256_hash(object_id):
+            raise ValueError(f"Invalid object ID format: {object_id}")
+
+        # Verify object_id and object_content match
+        if self.verify:
+            object_id_from_content = get_object_id(object_content)
+            if object_id != object_id_from_content:
+                raise ValueError(f"Object ID {object_id} does not match content hash")
```

Comment:
I wonder if this is something to do in the `*Servicer` that interfaces with the `ObjectStore` instead of doing it in here. In that way we can reuse these checks regardless of the `ObjectStore` implementation. 

I'm starting work on https://github.com/adap/flower/pull/5369 so I thought adding this to the `PushObject` method in the servicer is maybe a preferable alternative?